### ISSUE TITLE
 Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection

### ISSUE THREAD

**Issue by mferlay** [docker/for-win#611]
### Expected behavior
Download an image and run the container
### Actual behavior
Unable to download images
### Information

  - Diagnostic ID: 04FECEE9-EA0F-4558-9DD2-1DC576836119/2017-03-29_15-30-01
  - docker run hello-world without locally image
  - Windows 10
  - Docker
      Version 17.03.1-ce-win5 (10743)
      Channel: stable
      b18e2a5

### Steps to reproduce the behavior

  1. launch docker run hello-world without locally image in the least version of docker for windows

**Comment by hotdogee** [comment]
I'm also experiencing this issue.
Diagnostic ID: 79D6DCBD-03D0-46A5-93B8-A6A28E122565/2017-04-03_18-16-29

**Comment by mferlay** [comment]
I Lin, for my part, was only an issue with my network. Please, Check your own

 

De : Han Lin [mailto:notifications@github.com] 
Envoyé : lundi 3 avril 2017 12:21
À : docker/for-win <for-win@noreply.github.com>
Cc : mferlay <mferlay@gnubila.fr>; Author <author@noreply.github.com>
Objet : Re: [docker/for-win] Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (#611)

 

I'm also experiencing this issue.
Diagnostic ID: 79D6DCBD-03D0-46A5-93B8-A6A28E122565/2017-04-03_18-16-29

**Comment by mirzapourh** [comment]
I am experiencing the same issue since the latest update:
Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

**Comment by MaxPovver** [comment]
The same issue...

**Comment by IAmAStealer** [comment]
Same here,
Behind a proxy, installed 2 days ago, tried all the DNS / PROXY solutions already posted.

**Comment by friism** [comment]
I'm closing this issue since folks seem to have resolved the problem. @MaxPovver @hotdogee please open separate issues with diagnostic ids and details if you're still experiencing problems.

**Comment by videni** [comment]
any solution?

**Comment by Galilyou** [comment]
@videni try changing your DNS and proxy settings and retry. Changing the DNS from 8.8.8.8 to auto fixed it for me. Not sure why, though.

**Comment by DileepHariharanNokia** [comment]
Facing the same issue in windows enviornment. Any solutions?

**Comment by IAmAStealer** [comment]
I guess that you need to find the config file of docker on windows and add the Proxy. I didn't not check for windows but I guess that the problem is similar.
I can tell I added 'export http_proxy="http://myproxy:proxyport"' in the file /etc/default/docker on Linux.

**Comment by S4lem** [comment]
Same issue for me. I use Docker Quickstart Terminal

**Comment by S4lem** [comment]
I had the same issue on Windows.
I Fixed this by setting 8.8.8.8 as the default DNS for the VM. 

-  Go check your Network connections

-  Right click/Properties on your VM 

- Click once on Internet Protocol Version 4 (TCP/IPv4) to highlight it, then click on the Properties button again.

**Comment by fcrimins** [comment]
I was getting this same error in Windows 10 Home using Docker Toolbox. Changing my DNS to Google's 8.8.8.8 (secondary to 8.8.4.4) also worked for me (per @aasaanhai [here](https://github.com/moby/moby/issues/22635)). Detailed instructions for how to do so [here](https://www.cactusvpn.com/tutorials/how-to-set-up-smart-dns-on-windows-10/) (see section II).

**Comment by mriffault** [comment]
Thanks for the DNS tips, it works.
For your privacy, you may use a more neutral DNS server, not the Google one, visit https://www.opennic.org/ to find the closest open server ;)

**Comment by baltzarmattsson** [comment]
Restarting docker fixed it for me

**Comment by l3x** [comment]
In Ubuntu 16 had to change 
from: dns-nameservers 8.8.8.8 8.8.4.4
to: dns-nameservers 8.8.8.8
in /etc/network/interfaces file

**Comment by evinhas** [comment]
For anyone who is changing the DNS but it is not getting results, check the proxy.
You should setup a proxy for docker:
https://stackoverflow.com/questions/23111631/cannot-download-docker-images-behind-a-proxy

First, create a systemd drop-in directory for the docker service:

`mkdir /etc/systemd/system/docker.service.d`

Now create a file called /etc/systemd/system/docker.service.d/http-proxy.conf that adds the HTTP_PROXY environment variable:

```
[Service]
Environment="HTTP_PROXY=http://proxy.example.com:80/"
```

If you have internal Docker registries that you need to contact without proxying you can specify them via the NO_PROXY environment variable:

```
Environment="HTTP_PROXY=http://proxy.example.com:80/"
Environment="NO_PROXY=localhost,127.0.0.0/8,docker-registry.somecorporation.com"
```

Flush changes:

`$ sudo systemctl daemon-reload`

Verify that the configuration has been loaded:

```
$ sudo systemctl show --property Environment docker
Environment=HTTP_PROXY=http://proxy.example.com:80/
```

Restart Docker:

`$ sudo systemctl restart docker`

**Comment by s-dobrovolschi** [comment]
The following works for me. I'm using Boot2Docker on Windows 10:
Go to ~/.docker/machine/machines/default/config.json and add your own DNS server into HostOptions/EngineOptions/Dns and restart docker machine.
`{  
   "HostOptions": {
        "Driver": "",
        "Memory": 0,
        "Disk": 0,
        "EngineOptions": {
            "ArbitraryFlags": [],
            "Dns": ["8.8.8.8","8.8.4.4"] <-- set it here
        }
}`
https://stackoverflow.com/questions/34296230/how-to-change-default-docker-machines-dns-settings

**Comment by hilleer** [comment]
I met the same issue and solved it by **restarting docker**, as someone suggested here :-)

**Comment by andreipaz** [comment]
On Windows 10 docker's gui give some facilities.

Just right click on docker's icon in the tray bar and select "Settings" 

Then, on the Docker's window, select the "Network" section and change the DNS option from "Automatic" to "Fixed" and hit "Apply". Docker will restart itself after that. I putted the Google's DNS (8.8.8.8) and it worked fine to me.

Hope it helps.

**Comment by bfhobbes** [comment]
Both windows 10 pro machines I have here have the same issue ( using docker for windows )

It's always broken after a machine reboot, and fixed by using the docker for windows gui to restart docker. 
The issue is specific to docker as the machines work in all other ways. 
Its a major pain foor our  headless build machines which no longer just works from a boot and I have to remote desktop in to restart docker in order for it to function as a build client.

**Comment by mytech1235** [comment]
Make sure your proxy settings are correct in your docker settings. After I changed the proxy settings to Manual from "Use System Proxy" and supply the URL, it worked for me.

**Comment by ghayes7** [comment]
Still having this issue on Windows 10 with Docker v17.12.0-ce-win74 after an os boot / restart. No proxy but Docker still fails to connect. Restarting docker usually fixes the issue. Applying the DNS change in docker's network settings also fixes the issue but I'd wager it's because applying the network changes still enacts a docker restart.

**Comment by jonbgallant** [comment]
Some things that have worked:
1. Disconnect any VPN connections
2. Switch to Windows Containers and then back to Linux Containers

**Comment by WhatFreshHellIsThis** [comment]
I just downloaded Docker stable to my windows 10 today and ran into this issue, why on earth was it closed?

**Comment by starquake** [comment]
@WhatFreshHellIsThis The reason why it is closed is mentioned right before it is closed. Just scroll back...

**Comment by WhatFreshHellIsThis** [comment]
@starquake this should not have been closed in my opinion, nothing is "resolved" and this is the case where the issue was reported and this is the first result that comes up when I search for this problem.

**Comment by starquake** [comment]
@WhatFreshHellIsThis The issue was resolved. See comment by @mferlay. If you have a similar issue I suggest you do what @friism said: Open a new issue.

He probably needs to information to help you. Or maybe has other reasons. The point is: if you want results you should probably follow orders.

Hope this helps!

**Comment by hilleer** [comment]
@WhatFreshHellIsThis just to be sure - have you tryed to restart docker? :-) If you haven't read comments, that solved it for others.

**Comment by WhatFreshHellIsThis** [comment]
Yes, it was resolved (not by restarting, that did nothing) after I installed it tried it and it immediately failed and I needed it to work and had some free time on my hands so took the time to find a solution and after repeated searching came here to see it's a well known bug experienced by many people and apparently because there are some workarounds discovered by the victims of this bug it was decided there was no need to respect the users that took the time to report the issue and to respect the time of countless other potential users of the software that will give up on it after wasting their time by actually fixing it. 

It's baffling in the first place that such a showstopper bug would be considered fine to leave in there, but other people jumping in to excuse it and normalize it is to be honest even more baffling.

**Comment by IngoVals** [comment]
@WhatFreshHellIsThis How did you manage to resolve this?

**Comment by starquake** [comment]
@WhatFreshHellIsThis This incident was resolved that's why the issue was resolved. @friism suggested to anyone else still having the problem to create a new issue. Did you follow the instructions of @friism and create a new ticket with your details?

If you didn't, there is no reason for you to rant like that.

If you did and you were not helped, that's a whole different story.

Also please tell everyone how you fixed it.

**Comment by IngoVals** [comment]
I have found out that this is a known vpnkit bug but seems hard to find using google. This has been fixed but not released. Installing the edge version of docker for windows fixed this for me 

see:
https://github.com/Azure/azure-iot-pcs-remote-monitoring-dotnet/issues/62
https://github.com/docker/for-win/issues/1581
https://github.com/docker/for-win/issues/1432

This might be unrelated to the original issue here which I guess was a proxy issue but since google seems to point towards this post I think it's nice to leave this here at least.

tl;dr Uninstall _docker for windows_, install _docker for windows edge_ version. If you have images and containers [there are ways to save them between uninstall/install. ](https://docs.docker.com/docker-for-windows/faqs/#what-kind-of-feedback-are-we-looking-for)

**Comment by praneetloke** [comment]
@IngoVals installing the Edge version of Docker CE works with my company's proxy. Thanks for your heads-up!

**Comment by ThomasSchoenbeck** [comment]
I switched to the Edge channel within the Docker Settings menu. It basically did upgrade my current install, did not lose any settings or images. Now I can download images in Kitematic without issues.

Thanks @IngoVals for the solution.

**Comment by diegopintog** [comment]
Setting docker DNS manually to 8.8.8.8 and restarting it did the trik for me 👍

**Comment by heitorsilva** [comment]
For me, on Windows 10, the solution was closing Ubuntu window (WSL), closing Docker, reopening Docker and then reopening Ubuntu (WSL).

**Comment by sergio-domingues** [comment]
1st) Check if you are connected to Windows Containers
2nd) If not, Docker>Switch to Windows containers
3rd) After restart run > docker run hello-world

**Comment by jrschumacher** [comment]
[OSX / Mac OS] I had to manually set my docker proxy configuration via the Docker app. For some reason it wasn't respecting the system proxy settings. Monitoring CNTLM I wasn't seeing any traffic, but once I set the settings manually it lit up.

**Comment by judsonabraham301** [comment]
sometimes it will show this error when ur internet connection is not working properly.

**Comment by geerlingguy** [comment]
Fresh install on Windows 10, got the same error. Going to settings, then setting the Fixed DNS Server to 8.8.8.8 (it was on Auto), and applying the change got everything working again. Thanks for the helpful notes in this thread, not sure why Auto didn't work out of the box.

**Comment by velykov** [comment]
I've got this error today

**Comment by meotch** [comment]
Restart fixed the issue for me! Using Windows 10

**Comment by velykov** [comment]
So, I use google dns 8.8.8.8 and tor proxy, only this helps me.

**Comment by montao** [comment]
Doesn't work. Can't even add a proxy. This software is not ready for release with such a serious bug in it.

**Comment by velykov** [comment]
Use GUI for set proxy

**Comment by golvmopp** [comment]
_Switch to Windows containers_ seems to have worked for me, as suggested by sergio-domingues.

**Comment by encryptedcommerce** [comment]
I had the same issue with Kubuntu 18.04.  The fix that worked was adding an entry to /etc/hosts :
`54.243.34.90 quay.io`

**Comment by minakar** [comment]
i had the same issue with windows 10  so Setting docker DNS manually to 8.8.8.8 and restarting it did the trick for m
![14](https://user-images.githubusercontent.com/32802704/40724253-eb824a1a-6420-11e8-8d4f-b53395ed6cdc.PNG)

**Comment by AlanCarlyle** [comment]
I got the same issue and like many others tried numerous solutions, My environment is;

- Windows 10 Pro (resent and up-to-date)
- Docker version 18.03.1-ce, build 9ee9f40

Latest attempt to fix ths is **setting fixed DNS server** ....  
And do this **Worked**; thanks @minakar and others...

_I am having trouble believing something as basic, and fundamental to what docker is about, has been a problem for so long (I found mentions of it going back too 2016) and is still causing serious issues for people who are just [testing that docker can say "hello-world"](https://docs.docker.com/docker-for-windows/#test-your-installation)._

**Comment by yojeek** [comment]
Have this on mac os 10.13.4, thought that it was the issue with firewall but it's not. Docker deamon just stops to pulling images from network until restarted.

**Comment by garciasa** [comment]
For those who have this issue, make sure you are setting up the dns server which is in your company, not always google dns works behind the proxy, even if you can ping it inside your network, at least in my case.

**Comment by MrCroft** [comment]
Weird... for me, just restarting docker did the trick. Didn't have to change DNS Server to Fixed, but just restart Docker.
I guess that Docker just behaves funny some times, on Windows at least.
If I do a Ctrl+C while running `docker-compose up` for the first time on a project, before it finishes, then I can't do anything else until I restart Docker... this just as another weird example of how it behaves on Windows.

**Comment by velykov** [comment]
Last time me helps just restarting docker after start os

**Comment by laurin1** [comment]
I was not having this problem until recently and changing DNS to 8.8.8.8  is NOT a solution usable in corporate environments - we can't change our machines to point to external DNS, as that would break access to our local services.

**Comment by IAmAStealer** [comment]
> I was not having this problem until recently and changing DNS to 8.8.8.8 is NOT a solution usable in corporate environments - we can't change our machines to point to external DNS, as that would break access to our local services.

I was in corporate environment as well. if you can set the proxy to use your corporate environment it should do the trick. I posted the solution for Linux a bit up. If you can do it in windows you're welcome to post the solution.

**Comment by IAmAStealer** [comment]
Docker need to follow your corporate policy otherwise it won't be able to access internet to download wanted images. If you check your browser settings you should find proxy settings. Identify how your computer usually access internet and set docker to use the same path. Also set the same DNS that your OS uses.
Hope it helps.

**Comment by laurin1** [comment]
Actually, I changed them both to 8.8.8.8 and that part is working fine. We can use 8.8.8.8 on our local machines (I'm the network administrator as well :), but we can't access local resources with that DNS setting.

**Comment by laurin1** [comment]
We don't use a proxy, FYI, I'm just talking about other server services the containers need to access.

**Comment by IAmAStealer** [comment]
You can set up starting options for your container to use another network settings https://docs.docker.com/config/containers/container-networking/#dns-service or https://docs.docker.com/v17.09/engine/userguide/networking/default_network/configure-dns/
resolv.conf (Linux) in your container can set up two different DNS if required.

**Comment by laurin1** [comment]
Thanks.

**Comment by blaqniz** [comment]
I had the same issue and only found out after 30 minutes that I was on a VPN network for work which blocks other sites. I went off the VPN, and it worked :) This is definitely a network issue. When it said "not authenticated", I thought perhaps I needed some login credentials or so.

**Comment by JayaniSumudini** [comment]
> Restarting docker fixed it for me on Windows

this worked for me

**Comment by vpekarek** [comment]
Got the same problem. Changing password helped me. Looks like Docker has problem with some special character, eg. <->=]*:@?{

At least I have these symbols in my password.

**Comment by cagdas001** [comment]
Same case with @vpekarek. My password has had special characters.

Changing DNS to fixed (tried both Google and Yandex DNS) and restarting didn't help me. Changing the password solved for me

**Comment by phkrug** [comment]
> Got the same problem. Changing password helped me. Looks like Docker has problem with some special character, eg. <->=]*:@?{
> 
> At least I have these symbols in my password.

That did the trick. Make sure you use your username and not your email address to login.

**Comment by TomCoursow** [comment]
Changing password also fixed it for me...

**Comment by alefred** [comment]
> Same here,
> Behind a proxy, installed 2 days ago, tried many of the DNS / PROXY solutions already posted.
> 
> EDIT : http://stackoverflow.com/questions/23111631/cannot-download-docker-images-behind-a-proxy
> Repaired the proxy options for Docker with this link. Enjoy.

I haved the same problem with an additional special characters in proxy login, with this procedure download is done .

Take note:  %5C  is the encoding "\"(backslash) and %40 is the encoding "@" (at)

Thanks.

**Comment by banstola** [comment]
> @videni try changing your DNS and proxy settings and retry. Changing the DNS from 8.8.8.8 to auto fixed it for me. Not sure why, though.

Surprisingly , for me changing DNS to 8.8.8.8 worked.

**Comment by sreejin** [comment]
I am facing the same issue in win 10.

**Comment by super-kamil** [comment]
On Windows go to `settings > reset > reset to factory defaults...`

This will also recreate the network adapters and you should be able to pull again.

**Comment by PrasannaMondkar** [comment]
@Galilyou 
Thanks. DNS Change to 8.8.8.8 in Docker Network settings worked for me as well.

**Comment by muralib91dotnet** [comment]
@Galilyou @PrasannaMondkar 
Thank you all. Changing DNS to 8.8.8.8 in Docker network settings worked for me too.

**Comment by emrecavunt** [comment]
it's weird for me also. I just CTRL + C on windows machine before run docker build. And it's just worked as a charm :/

**Comment by sreetejarusum** [comment]
I tried changing DNS to 8.8.8.8 in Docker network settings and restarted my docker, i am still not able to download images.

**Comment by junghanChoi** [comment]
Just restart docker solved my problem.

**Comment by SBlechmann** [comment]
Well, for me the issue was that I forgot to give docker access to my local drive <.<*

**Comment by houmanb** [comment]
WE are actually sitting behind our company proxy and this configuration worked at least for us:
/etc/systemd/system/docker.service.d/http-proxy.conf

```
[Service]
Environment="http_proxy=http://proxy.xxxxxx:8080/"
Environment="https_proxy=http://proxy.xxxxxx:8080/"
Environment="HTTP_PROXY=http://proxy.xxxxxx:8080/"
Environment="HTTPS_PROXY=http://proxy.xxxxxx:8080/"
```

**Comment by markelliottsmarsh** [comment]
Thanks. DNS Change to 8.8.8.8 in Docker Network settings worked for me as well.

**Comment by jyleehe** [comment]
@andreipaz Thanks It works!

**Comment by sbabaei** [comment]
Changing to Windows Containers is worked for but, But I want to use Linux Containers too.

**Comment by svenluijten** [comment]
Chiming in here for people still running into this issue: 

You should run `docker login` from _outside_ of WSL. I used PowerShell, and it worked first time.

**Comment by brabia** [comment]
Facing the same issue in ubuntu 16.04 enviornment, no proxy, any solutions?

**Comment by brabia** [comment]
SOLUTION:
ubuntu 16.04
no proxy

Updated /etc/resolv.conf
nameserver 10.0.2.3
nameserver 8.8.8.8
nameserver 8.8.4.4

**Comment by Tiantiandas** [comment]
restarting docker also works for me.

**Comment by AlbertSugi** [comment]
> it's weird for me also. I just CTRL + C on windows machine before run docker build. And it's just worked as a charm :/

WOW! This is like magic! Thanks a lot!

**Comment by AlbertSugi** [comment]
> Chiming in here for people still running into this issue:
> 
> You should run `docker login` from _outside_ of WSL (1.0). I used PowerShell, and it worked first time.

This works for me Thank you a lot.

**Comment by andyraddatz** [comment]
> Restarting docker fixed it for me on Windows

I'm on Windows 10 Pro, none of the DNS things fixed it or restarting docker, but restarting my PC did fix it.

**Comment by FrostyJet** [comment]
In my case, removing registry-1.docker.io from my hosts files fixed the issue. (running on windows)

**Comment by montao** [comment]
I had a strange error message trying to run docker behind a proxy last year, turned out that I had a typo in my proxy conf, had to fix "`http:/proxy..`" and correct it "`http://proxy...`". It took some time to find because the error message was not really informative...

**Comment by jiovanniro** [comment]
> Restarting docker fixed it for me on Windows

If you're not sure how to restart docker, this forum was helpful:
https://forums.docker.com/t/restart-docker-service-from-command-line/27331/6

`docker-machine restart
`

**Comment by komali2** [comment]
I fixed this issue by continuing to repeat my `docker-compose up` command, but on one of my attempts I also ran the google speed test, which i think spooked comcast into giving me the network speed I needed to not timeout. Then it worked.

**Comment by AvirupDas** [comment]
Updating nameservers to 8.8.8.8 within /etc/resolv.conf fixed the issue.

**Comment by Danielotuo** [comment]
restarting docker fixed the issue

`docker-machine restart`

**Comment by MateoLa** [comment]
On linux, and working with rails, ruby verison on Dockerfile must match ruby version on Gemfile.
And thats it.

**Comment by lucassutelo** [comment]
Alterando o DNS para o do Google: 8.8.4.4 funcionou para mim.
